java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -4950,4 +4951,92 @@ public void sintercard(BaseClient client) {
+        Transaction setFoobarTransaction = new Transaction();
+        Transaction setHelloTransaction = new Transaction();
+        String[] expectedExecResponse = new String[] {OK, OK, OK};
+
+        // Returns null when a watched key is modified before it is executed in a transaction command.
+        // Transaction commands are not performed.
+        assertEquals(OK, standaloneClient.watch(keys).get());
+        assertEquals(OK, standaloneClient.set(key2, helloString).get());
+        setFoobarTransaction.set(key1, foobarString).set(key2, foobarString).set(key3, foobarString);
+        assertEquals(null, standaloneClient.exec(setFoobarTransaction).get());
+        assertEquals(null, standaloneClient.get(key1).get());
+        assertEquals(helloString, standaloneClient.get(key2).get());
+        assertEquals(null, standaloneClient.get(key3).get());
+
+        // Transaction executes command successfully with a read command on the watch key before
+        // transaction is executed.
+        assertEquals(OK, standaloneClient.watch(keys).get());
+        assertEquals(helloString, standaloneClient.get(key2).get());
+        assertArrayEquals(expectedExecResponse, standaloneClient.exec(setFoobarTransaction).get());
+        assertEquals(foobarString, standaloneClient.get(key1).get());
+        assertEquals(foobarString, standaloneClient.get(key2).get());
+        assertEquals(foobarString, standaloneClient.get(key3).get());
+
+        // Transaction executes command successfully with an unmodified watched key
+        assertEquals(OK, standaloneClient.watch(keys).get());
+        assertArrayEquals(expectedExecResponse, standaloneClient.exec(setFoobarTransaction).get());
+        assertEquals(foobarString, standaloneClient.get(key1).get());
+        assertEquals(foobarString, standaloneClient.get(key2).get());
+        assertEquals(foobarString, standaloneClient.get(key3).get());
+
+        // Transaction executes command successfully with a modified watched key but is not in the
+        // transaction.
+        assertEquals(OK, standaloneClient.watch(new String[] {key4}).get());
+        setHelloTransaction.set(key1, helloString).set(key2, helloString).set(key3, helloString);
+        assertArrayEquals(expectedExecResponse, standaloneClient.exec(setHelloTransaction).get());
+        assertEquals(helloString, standaloneClient.get(key1).get());
+        assertEquals(helloString, standaloneClient.get(key2).get());
+        assertEquals(helloString, standaloneClient.get(key3).get());
+
+        // Transaction executes command successfully with an unmodified watched key
+        assertEquals(OK, standaloneClient.watch(keys).get());
+        assertArrayEquals(expectedExecResponse, standaloneClient.exec(setFoobarTransaction).get());
+        assertEquals(foobarString, standaloneClient.get(key1).get());
+        assertEquals(foobarString, standaloneClient.get(key2).get());
+        assertEquals(foobarString, standaloneClient.get(key3).get());
+
```

Comment:
do we want to have a test where we watch a non-existing key, set the non-existing key, try and execute the transaction, and assert that it was aborted? not sure if its necessary or not

Reply:
I think that's the first test. Is that what you meant?